lib/hammer_cli_foreman/compute_resource.rb

```diff
@@ -70,17 +70,29 @@ def print_data(data)
 
     class CreateCommand < HammerCLIForeman::CreateCommand
 
+      VALIDATION_PER_PROIVIDER = {
+        'libvirt'   => [:option_url],
+        'ovirt'     => [:option_url, :option_user, :option_password, :option_datacenter],
+        'openstack' => [:option_url, :option_user, :option_password],
+        'racksapce' => [:option_url],
+        'ec2'       => [:option_region, :option_user, :option_password],
+        'vmware'    => [:option_user, :option_password, :option_datacenter, :option_server]
+      }
+
       success_message _("Compute resource created")
       failure_message _("Could not create the compute resource")
 
       build_options
 
       validate_options do
-        all(:option_name, :option_url, :option_provider).required
+        all(
+          *VALIDATION_PER_PROIVIDER[
+            option(:option_provider).value.to_s.downcase
+          ].unshift(:option_name, :option_provider)
```

Comment:
I get an error when I try to create a compute resource without specifying the provider:
```
hammer compute-resource create --name Test
Could not create the compute resource:
  Error: undefined method `unshift' for nil:NilClass
```
the same happens when I use some unexpected value for the provider, eg. `--provider=unknown`

Reply:
@tstrachota Thanks for reviewing. I have fixed the above error. Functional test added.